src/transformers/data/data_collator.py

```diff
@@ -313,3 +314,186 @@ def mask_tokens(self, inputs: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor,
             ) & masked_indices[i]
 
         return inputs, perm_mask, target_mapping, labels
+
+
+@dataclass
+class DataCollatorForNextSentencePrediction:
+    """
+    Data collator used for language modeling.
+    - collates batches of tensors, honoring their tokenizer's pad_token
+    - preprocesses batches for masked language modeling
+    """
+
+    tokenizer: PreTrainedTokenizer
+    mlm: bool = True
+    block_size: int = 128
+    short_seq_probability: float = 0.1
+    nsp_probability: float = 0.5
+    mlm_probability: float = 0.15
+
+    def __call__(self, examples: List[Union[List[List[int]], Dict[str, torch.Tensor]]]) -> Dict[str, torch.Tensor]:
+        if isinstance(examples[0], (dict, BatchEncoding)):
+            examples = [e["input_ids"] for e in examples]
```

Comment:
Sorry, I'm a little confused.
The `labels` you mentioned are nsp/mlm labels, or labels for a specific task?
Since none of data collators in this file grab the `token_type_ids` and `labels`, they just take the examples out of the dict, and do nothing else. 
Thank you~

Reply:
Ah sorry, I was reading this wrong.

Reply:
aha, ok~ :-)